mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -233,7 +233,7 @@ class Word2Vec extends Serializable with Logging {
       a += 1
     }
     while (a < 2 * vocabSize) {
-      count(a) = 1e9.toInt
+      count(a) = Int.MaxValue
```

Comment:
Both values are different. Why do we need to change? Can you file a JIRA since before/after aren't virtually same.

```scala
scala> Int.MaxValue
res0: Int = 2147483647

scala> 1e9.toInt
res1: Int = 1000000000
```

Reply:
if VocabWord.cn > 1e9.toInt and  VocabWord.cn < Int.MaxValue,  the createBinaryTree is error;
Causes an infinite loop of line 283；

Reply:
If not mistaken the array holds longs should not be a Long Max? my 2 cents.

Reply:
I think it can't be Long.MaxValue as later it sums counts which, I guess, could overflow. This is just supposed to be "a very big value" and I think it's safe to assume none are over Int.MaxValue. I suppose we could cap the values in the vocab array to this value to be safe. So, I think the change is OK.

Reply:
For reference, in an early implementation of word2vec, it uses 1e15: https://github.com/tmikolov/word2vec/blob/master/word2vec.c#L214

Reply:
@wangbin0227 can you file a JIRA?

Reply:
PS @skonto you are correct. Given how this works, `Long.MaxValue` is fine as those values are never added. I took this over in https://github.com/apache/spark/pull/24893